src/hotspot/share/opto/predicates.cpp

```diff
@@ -381,6 +382,83 @@ bool TemplateAssertionPredicateExpressionNode::is_template_assertion_predicate(N
   return node->is_If() && node->in(1)->is_Opaque4();
 }
 
+InitializedAssertionPredicate::InitializedAssertionPredicate(IfNode* template_assertion_predicate, Node* new_init,
+                                                             Node* new_stride, PhaseIdealLoop* phase)
+    : _template_assertion_predicate(template_assertion_predicate),
+      NOT_PRODUCT(_assertion_predicate_type(template_assertion_predicate->assertion_predicate_type()) COMMA)
+      _new_init(new_init),
+      _new_stride(new_stride),
+      _phase(phase) {}
+
+// Create an Initialized Assertion Predicate at the provided control from the _template_assertion_predicate:
+//
+//          init                 stride
+//            |                    |
+//   OpaqueLoopInitNode  OpaqueLoopStrideNode                      _new_init    _new_stride
+//                  \     /                                               \     /
+//                    ...                                                   ...
+//                     |                                                     |
+//                    Bool                                                new Bool
+//                     |                                                     |
+//                  Opaque4                ======>              control   OpaqueInitializedAssertionPredicate
+//                     |                                              \  /
+//                    If                                             new If
+//                  /    \                                           /    \
+//            success     fail path                         new success   new Halt
+//              proj    (Halt or UCT)                           proj
+//
+IfTrueNode* InitializedAssertionPredicate::create(Node* control) {
+  IdealLoopTree* loop = _phase->get_loop(control);
+  OpaqueInitializedAssertionPredicateNode* new_opaque_bool = create_new_bool(control);
+  IfNode* if_node = create_if_node(control, new_opaque_bool, loop);
+  create_halt_path(if_node, loop);
+  return create_success_path(if_node, loop);
+}
+
+// Create a OpaqueInitializedAssertionPredicate with a new Assertion Predicate Expression.
+OpaqueInitializedAssertionPredicateNode* InitializedAssertionPredicate::create_new_bool(Node* control) {
+  Opaque4Node* template_opaque_bool = _template_assertion_predicate->in(1)->as_Opaque4();
+  TemplateAssertionPredicateExpression template_assertion_predicate_expression(template_opaque_bool);
+  Opaque4Node* new_opaque =
+      template_assertion_predicate_expression.clone_and_replace_init_and_stride(_new_init, _new_stride, control, _phase);
+  OpaqueInitializedAssertionPredicateNode* opaque_bool
+      = new OpaqueInitializedAssertionPredicateNode(new_opaque->in(1)->as_Bool(), _phase->C);
+  _phase->register_new_node(opaque_bool, control);
+  return opaque_bool;
```

Comment:
```suggestion
  Opaque4Node* template_opaque = _template_assertion_predicate->in(1)->as_Opaque4();
  TemplateAssertionPredicateExpression expression(template_opaque);
  Opaque4Node* tmp_opaque = expression.clone_and_replace_init_and_stride(_new_init, _new_stride, control, _phase);
  BoolNode* new_bool = tmp_opaque->in(1)->as_Bool();
  OpaqueInitializedAssertionPredicateNode* new_opaque = new OpaqueInitializedAssertionPredicateNode(new_bool, _phase->C);
  _phase->register_new_node(new_opaque, control);
  return new_opaque;
```

`template_opaque_bool` -> `template_opaque` (it is not a bool!)

`template_assertion_predicate_expression` -> `expression` (clear enough given type)

`new_opaque` -> `tmp_opaque` (this is a temporary, right?)

`opaque_bool` -> `new_opaque`

Suggested name for this method: `create_initialized_assertion_predicate_expression`.

Reply:
See comment below.